### ISSUE TITLE
unable to log in if cryptpad is behind authelia - NPM

### ISSUE THREAD

**Issue by GravityEcho** [cryptpad/cryptpad#1599]
### Description

Unable to log in if cryptpad is behind authelia (with docker and NGINX Proxy Manager). Everything works well until I tell NPM to put authelia in front of cryptpad, then the login "hashing password" never finishes.

### Steps to reproduce

1. install cryptpad on docker with NPM and authelia
2. create account with cryptpad, secure dns with ssl
3. put authelia in front of cryptpad
4. unable to log in to cryptpad (hashing password never finishes)

### Expected behavior

can still log in to cryptpad after successful authentication with authelia

### Which operating system are you using?

Linux/BSD/UNIX

### What browsers are you seeing the problem on?

Google Chrome

### Extensions

N/A

### Version

2024.6.1

**Comment by svenEsven** [comment]
I've been tinkering at trying to get this up and running in docker without authelia and having the same issues. Are you sure the issues aren't there without authelia?

**Comment by GravityEcho** [comment]
> I've been tinkering at trying to get this up and running in docker without authelia and having the same issues. Are you sure the issues aren't there without authelia?
> 
> if not i would appreciate any steps you took to get it there.

Yes, to illustrate, If I try to log in when cryptpad is behind authelia, it just basically freezes at "hashing password" at login. While it is doing that, if I remove the custom nginx config that puts authelia in front of cryptpad, it proceeds to log in, no restart or anything else needed.

It could very well be my custom config but it works for all 12 of my other web-apps. Custom config follows:

location /authelia {
    internal;
    set $upstream_authelia http://192.168.0.244:9091/api/verify; #ADD YOUR IP AND PORT OF AUTHELIA
    proxy_pass_request_body off;
    proxy_pass $upstream_authelia;    
    proxy_set_header Content-Length "";
 
    # Timeout if the real server is dead
    proxy_next_upstream error timeout invalid_header http_500 http_502 http_503;
    client_body_buffer_size 128k;
    proxy_set_header Host $host;
    proxy_set_header X-Original-URL $scheme://$http_host$request_uri;
    proxy_set_header X-Real-IP $remote_addr;
    proxy_set_header X-Forwarded-For $remote_addr; 
    proxy_set_header X-Forwarded-Proto $scheme;
    proxy_set_header X-Forwarded-Host $http_host;
    proxy_set_header X-Forwarded-Uri $request_uri;
    proxy_set_header X-Forwarded-Ssl on;
    proxy_redirect  http://  $scheme://;
    proxy_http_version 1.1;
    proxy_set_header Connection "";
    proxy_cache_bypass $cookie_session;
    proxy_no_cache $cookie_session;
    proxy_buffers 4 32k;
 
    send_timeout 5m;
    proxy_read_timeout 240;
    proxy_send_timeout 240;
    proxy_connect_timeout 240;
}
 
    location / {
        set $upstream_cryptpad http://192.168.0.243:3000;  #ADD IP AND PORT OF SERVICE
        proxy_pass $upstream_cryptpad;  #change name of the service
 
        auth_request /authelia;
        auth_request_set $target_url $scheme://$http_host$request_uri;
        auth_request_set $user $upstream_http_remote_user;
        auth_request_set $groups $upstream_http_remote_groups;
        proxy_set_header Remote-User $user;
        proxy_set_header Remote-Groups $groups;
        error_page 401 =302 http://authelia.example.com/?rd=$target_url;
 
        client_body_buffer_size 128k;
 
        proxy_next_upstream error timeout invalid_header http_500 http_502 http_503;
 
        send_timeout 5m;
        proxy_read_timeout 360;
        proxy_send_timeout 360;
        proxy_connect_timeout 360;
 
        proxy_set_header Host $host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;
        proxy_set_header X-Forwarded-Host $http_host;
        proxy_set_header X-Forwarded-Uri $request_uri;
        proxy_set_header X-Forwarded-Ssl on;
        proxy_redirect  http://  $scheme://;
        proxy_http_version 1.1;
        proxy_set_header Connection "";
        proxy_cache_bypass $cookie_session;
        proxy_no_cache $cookie_session;
        proxy_buffers 64 256k;
 
   
        set_real_ip_from 192.168.0.0/16;
        #set_real_ip_from 192.168.0.0/8;
        real_ip_header X-Forwarded-For;
        real_ip_recursive on;
    }

**Comment by GravityEcho** [comment]
double checked just now and yep works fine behind NPM with SSL at my domain. Here is my docker compose if it helps:
For testing I'm not bothering with onlyoffice it's got it's own mess of issues.

services:
  cryptpad:
    image: cryptpad/cryptpad:version-2024.6.1
    container_name: cryptpad
    hostname: cryptpad

    environment:
      - CPAD_MAIN_DOMAIN=https://cryptpad.example.com
#      - CPAD_MAIN_DOMAIN=http://localhost:3000
#      - CPAD_SANDBOX_DOMAIN=https://your-sandbox-domain.com
      - CPAD_CONF=/cryptpad/config/config.js
      # Read and accept the license before uncommenting the following line:
      # https://github.com/ONLYOFFICE/web-apps/blob/master/LICENSE.txt
#      - CPAD_INSTALL_ONLYOFFICE=yes


    volumes:
      - ./data/blob:/cryptpad/blob
      - ./data/block:/cryptpad/block
      - ./customize:/cryptpad/customize
      - ./data/data:/cryptpad/data
      - ./data/files:/cryptpad/datastore
#      - ./onlyoffice-dist:/cryptpad/www/common/onlyoffice/dist
#      - ./onlyoffice-conf:/cryptpad/onlyoffice-conf

    ports:
      - "3000:3000"
      - "3003:3003"

    ulimits:
      nofile:
        soft: 1000000
        hard: 1000000
